### PR TITLE
[PATCH v4] linux-gen: pktio: use names other than index also as dpdk ports

### PR DESCRIPTION
**Pull request by vijayinavolu**
dpdk_open() only allows netdev string to be a port number
This is to allow passing all other supported DPDK ports as well.

Signed-off-by: Vijay Ram Inavolu <vinavolu@marvell.com>

### CODE REVIEW

**Comment by vijayinavolu** [comment]
Related odp-dpdk PR
https://github.com/OpenDataPlane/odp-dpdk/pull/135

File: platform/linux-generic/pktio/dpdk.c [hunk 1]
```diff
@@ -1684,8 +1685,12 @@ static int dpdk_open(odp_pktio_t id ODP_UNUSED,
 		return -1;
 	pool_entry = pool_entry_from_hdl(pool);
 
-	if (!dpdk_netdev_is_valid(netdev)) {
-		ODP_ERR("Invalid dpdk netdev: %s\n", netdev);
+	if (!rte_eth_dev_get_port_by_name(netdev, &port_id))
+		pkt_dpdk->port_id = port_id;
+	else if (dpdk_netdev_is_valid(netdev))
+		pkt_dpdk->port_id = atoi(netdev);
```
Comment (MatiasElo) [inline platform/linux-generic/pktio/dpdk.c:1691]: There is a `memset()` below which initializes `pkt_dpdk `.

File: platform/linux-generic/pktio/dpdk.c [hunk 2]
```diff
@@ -1684,8 +1685,12 @@ static int dpdk_open(odp_pktio_t id ODP_UNUSED,
...
+	else {
+		ODP_ERR("Invalid interface name!: %s\n", netdev);
```
Comment (MatiasElo) [inline platform/linux-generic/pktio/dpdk.c:1693]: In linux-generic implementation a user may want to use also other interface types than DPDK, so maybe it  would be better to print:
`ODP_DBG("Invalid DPDK interface name: %s\n", netdev);`

File: platform/linux-generic/pktio/dpdk.c (hunk 1 above)
Comment (vijayinavolu) [reply platform/linux-generic/pktio/dpdk.c:1691]: moved the memset up

File: platform/linux-generic/pktio/dpdk.c (hunk 2 above)
Comment (vijayinavolu) [reply platform/linux-generic/pktio/dpdk.c:1693]: Updated the print statement

File: platform/linux-generic/pktio/dpdk.c [hunk 3]
```diff
@@ -1684,8 +1685,15 @@ static int dpdk_open(odp_pktio_t id ODP_UNUSED,
...
-		ODP_ERR("Invalid dpdk netdev: %s\n", netdev);
+	/* Init pktio entry */
+	memset(pkt_dpdk, 0, sizeof(*pkt_dpdk));
+
+	if (!rte_eth_dev_get_port_by_name(netdev, &port_id))
+		pkt_dpdk->port_id = port_id;
+	else if (dpdk_netdev_is_valid(netdev))
+		pkt_dpdk->port_id = atoi(netdev);
+	else {
+		ODP_ERR("Invalid DPDK interface name!: %s\n", netdev);
```
Comment (MatiasElo) [inline platform/linux-generic/pktio/dpdk.c:1696]: You can still remove the exclamation mark but otherwise the PR is looking good.

**Review by MatiasElo — APPROVED** [review]

File: platform/linux-generic/pktio/dpdk.c (hunk 3 above)
Comment (vijayinavolu) [reply platform/linux-generic/pktio/dpdk.c]: Changed the log message and added reviewed-by tag

**Pull request merged**